v-next/core/src/internal/providers/errors.ts

```diff
@@ -0,0 +1,57 @@
+import type { ProviderRpcError } from "../../types/providers.js";
```

Comment:
Everything networks related should be in hardhat, not core. Ideally, inside a folder like `v-next/hardhat/src/internal/network`.

Reply:
Done [here](https://github.com/NomicFoundation/hardhat/pull/5580/commits/8f13be53096fc3eea0191fef0b4b28e3312124ce)